### ISSUE TITLE
rustc_llvm need to be rebuilt after ./x.py check

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#71152]
repo @ edc02580e4e80476ac1ded2cc1008eaf8b8400e6
running

```
./x.py build
./x.py check
```
invalidates the build cache for the `rustc_llvm` crate.
This means that after
````
./x.py build
./x.py check
./x.py build
````
the last `x.py build` will have to rebuild `rustc_llvm`, `rustc_codegen_llvm`, `rustc_interface`, `rustc_driver` and `rustc-main` which is less than optional given no code was changed.

**Comment by cuviper** [comment]
I think it's because of this:
https://github.com/rust-lang/rust/blob/edc02580e4e80476ac1ded2cc1008eaf8b8400e6/src/bootstrap/builder.rs#L767-L771
https://github.com/rust-lang/rust/blob/edc02580e4e80476ac1ded2cc1008eaf8b8400e6/src/librustc_llvm/build.rs#L18-L22

**Comment by Mark-Simulacrum** [comment]
Interesting. Ideally we'd have two separate caches for rustc_llvm and the other crates such that x.py check doesn't affect this... I think.

**Comment by jonas-schievink** [comment]
Is there any workaround for this other than not using `./x.py check`?

**Comment by Mark-Simulacrum** [comment]
Fixed in #71815
